core/chains/evm/gas/block_history_estimator.go

```diff
@@ -128,11 +127,11 @@ func NewBlockHistoryEstimator(lggr logger.Logger, ethClient evmclient.Client, cf
 
 // OnNewLongestChain recalculates and sets global gas price if a sampled new head comes
 // in and we are not currently fetching
-func (b *BlockHistoryEstimator) OnNewLongestChain(_ context.Context, head *evmtypes.Head) {
+func (b *BlockHistoryEstimator) OnNewLongestChain(_ context.Context, head txmgrtypes.Head) {
 	// set latest base fee here to avoid potential lag introduced by block delay
 	// it is really important that base fee be as up-to-date as possible
-	b.setLatest(head)
-	b.mb.Deliver(head)
+	b.setLatest(head.(*evmtypes.Head))
+	b.mb.Deliver(head.(*evmtypes.Head))
```

Comment:
Ah so here is one place where a generic data type could avoid a runtime cast. Would it be possible to make just the `HeadTrackable` interface use a generic type, without it spreading everywhere else?
```go
type HeadTrackable[H Head] interface {
	OnNewLongestChain(ctx context.Context, head H)
}
```

Reply:
++1 to this!

I think it could just be
```go
HeadTrackable[H txmgrtypes.Head]
```
(allows generic type, but type must also conform to expected interface)

Reply:
I tried that exact thing in my previous, now obsolete PR. The Head item is actually stored inside a mailbox in TxMgr, and EthConfirmer. So doing this also requires TxMgr[Head any] and EthConfirmer[Head any] to be added.
That blows up the change more widely.

Also, I think, even if we do that, @samsondav is generally in favor of using interfaces over generics, except when there's no way around that.

Reply:
I'm not sure I follow all the steps from this high of a level, but avoiding a runtime cast that can panic in favor of compile time type safety is a textbook case for generics. When would we use them, if not for this? Is there some other way to avoid the cast here?